src/linkage.md

```diff
@@ -301,8 +298,9 @@ r[link.unwinding.prohibited]
 If a Rust artifact is potentially unwinding, then all its crates must be built with the `unwind` [panic strategy].
 
 > [!NOTE]
-> This restriction can only be violated when mixing code with different [`-C panic`] flags
-> while also using a non-`rustc` linker. Most users do not have to be concerned about this.
+> If you are *not* using `rustc` to link, you must take care to ensure that unwinding is handled consistently across the entire binary. Linking without `rustc` includes using `dlopen` or similar facilities where linking is done by the system runtime without `rustc` being involved.
```

Comment:
I think it'd still be good to say explicitly that most users don't have to worry about this.
```suggestion
> If you are using `rustc` to link, these rules are enforced automatically.
> If you are *not* using `rustc` to link, you must take care to ensure that unwinding is handled consistently across the entire binary. Linking without `rustc` includes using `dlopen` or similar facilities where linking is done by the system runtime without `rustc` being involved.
```